### ISSUE TITLE
Failing verification when using td.object() as argument

### ISSUE THREAD

**Issue by lumaxis** [testdouble/testdouble.js#426]
### Description

When passing a `td.object()` as a parameter to a function that then call a callback with that function, verifying the invocation on that callback fails.
I attached an example that repros the issue in the repl and is fairly similar to my real world case. It's very possible that this is also just me misusing the library but I imagine it's fairly easy to fall into this trap? 🤔

Additionally, this is possibly a duplicate of #419 but I decided to open a separate issue instead since this is when using `td.verify()`, not `td.when()`.

### Issue

Testdouble fails to correctly verify the invocation even though the function was seemingly correctly called. Additionally, the `(undefined)` name in the error message is confusing as it makes it very hard to understand what exactly the problem is/in what way the arguments don't match.

```
Error: Unsatisfied verification on test double `logger`.

  Wanted:
    - called with `(undefined)`.

  All calls of the test double, in order were:
    - called with `(undefined)`.
    at Object.exports.default [as verify] (/Users/lukas/code/testdouble.js/src/verify.js:17:9)
    at Object.fail (/Users/lukas/code/testdouble.js/src/log.js:17:11)
```

### Environment

- [x] `node -v` output: `v10.11.0`
- [x] `npm -v` (or `yarn --version`) output: `6.4.1`
- [x] `npm ls testdouble` (or `yarn list testdouble`) version: `3.12.3`

#### Code-fenced Examples

```js
function middlewareFactory(logger) {
  return (res) => {
    if (logger) { logger(res); }
  };
}

const fakeResponse = td.object('response');
const fakeLogger = td.function('logger');

const middleware = middlewareFactory(fakeLogger);
middleware(fakeResponse);

td.verify(fakeLogger(fakeResponse));
```

**Comment by searls** [comment]
Is this a dupe of #419?

I'd really appreciate the following help:
- [ ] ascertain if this is the same issue (I suspect it is, since the code that determines if `args-match` is the same for stubbing & verifying)
- [ ] write a short test script and do a git bisect to see if this was introduced recently—I suspect it was—and identify the offending commit
- [ ] reply with any thoughts on a fix

**Comment by lumaxis** [comment]
@searls Sure! I took another closer look at this.

- I now **do** believe this is the same as #419.
     The issue that the assertion does fail at all is indeed from `isMatcher` returning `true` for the `td.object()` here: https://github.com/testdouble/testdouble.js/blob/b2dfa19e959608be2821968fcd1ef62936b61466/src/args-match.js#L22-L23
    However, I do not believe it was introduced recently, this seems to have been failing for a long while.
- The error message being broken with `(undefined)` seems to have been introduced in https://github.com/testdouble/testdouble.js/commit/c59e4fde43f9747ba5feb77f8e5c9a550ee114b3

    **Before**:
    ```js
            throw new Error(msg)
              ^
    Error: Unsatisfied verification on test double `logger`.
    
      Wanted:
        - called with `([test double for "response.__name"])`.
    
      All calls of the test double, in order were:
        - called with `([test double for "response.__name"])`.
    ```
    **After**:
    ```js
        throw new Error(msg)
              ^
    Error: Unsatisfied verification on test double `logger`.
    
      Wanted:
        - called with `(undefined)`.
    
      All calls of the test double, in order were:
        - called with `(undefined)`.
     ```

As for potential fixes, I don't have any immediate ideas but I'll take another look tonight.

**Comment by lumaxis** [comment]
Added a suggested fix in #427. @searls Please take a look and let me know what you think 🙂
